### PR TITLE
FEAT : Added the update testimonial endpoint

### PR DESCRIPTION
**Pull request by marveldo**
> > #FEAT : Added a protected endpoint to create testimonials
> > ## Description
> > This PR adds a new endpoint In which Users can Update their testimonials if needed be it requires the users to be authenticated and checks if the user is the owner Ive made sure to not touch the migrations in this pull request
> > ## Related Issue (This is the Link to issue ticket)
> > [[Link to issue ticket](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/144)]
> > ## Motivation and Context
> > This change is required to allow authenticated users to change their respective testimonials
> > ## How Has This Been Tested?
> > 
> > * This Update testimonial endpoint was tested using POSTMAN ,API SWAGGERUI and PYTEST .
> > * Verified that only authenticated users can access the endpoint.
> > * Tested the endpoint with different user valid and invalid data to ensure it was performing to standards
> > * Ensured that the testimonial is correctly updated in the database
> > 
> > ## Screenshots (if appropriate - Postman, etc):
> > pytest Request
> > ## Types of changes
> > 
> > * [x]   New feature (A non breaking functionality change which cintributes to the api)
> > 
> > ## Checklist:
> > 
> > * [x]   My code follows the code style of this project.
> > * [ ]   My change requires a change to the documentation.
> > * [ ]   I have updated the documentation accordingly.
> > * [x]   I have read the **CONTRIBUTING** document.
> > * [x]   I have added tests to cover my changes.
> > * [x]   All new and existing tests passed.
> 
> ​

​

## Motivation and Context

​

## How Has This Been Tested?

​

## Screenshots (if appropriate - Postman, etc):

​

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: api/v1/models/testimonials.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+from sqlalchemy.dialects.postgresql import UUID       
+
+class Testimonial(BaseModel, Base):  
+    __tablename__ = 'testimonials'     
+    
+    firstname = Column(String(50), nullable=False)  
+    lastname = Column(String(50), nullable=False)   
+    content = Column(Text, nullable=False)    
+    user_id = Column(UUID(as_uuid=True), ForeignKey('users.id'))    
+    user = relationship("User", backref="testimonials")
```
Comment (Dev-wonderful) [inline api/v1/models/testimonials.py:21]: Where's the migration for this

File: api/v1/routes/testimonials.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+async def custom_exception_handler(request: Request, exc: CustomException):
+    return JSONResponse(
+        status_code=exc.status_code,
+        content={
+            "message": exc.message,
+            "success": exc.success,
+            "status_code": exc.status_code
+        }
+    )
+
```
Comment (Dev-wonderful) [inline api/v1/routes/testimonials.py:50]: Check out the `JsonResponseDict` class in `utils.json_response` module

File: api/v1/routes/testimonials.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+BaseTestimonialResponse
+
+)
+from api.db.database import get_db, Base, engine
+from typing import Annotated
+from api.utils.dependencies import get_current_user
+
+
+Base.metadata.create_all(bind=engine)
+
```
Comment (Dev-wonderful) [inline api/v1/routes/testimonials.py:26]: What is this doing here

**Review by Dev-wonderful — CHANGES_REQUESTED** [review]

File: tests/v1/testupdatetestimonial.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    }
+    data = {
+        'content': 'I love doyin'
+    }
+
+    response = requests.put(url, headers=headers, data=json.dumps(data))
+
+    assert response.status_code == 200
+
+    assert "data" in response.json()
```
Comment (Dev-wonderful) [inline tests/v1/testupdatetestimonial.py:24]: Do you think this test would pass without this server running `http://127.0.0.1:7000`

**Review by Dev-wonderful — CHANGES_REQUESTED** [review]

File: api/v1/models/testimonials.py (hunk 1 above)
Comment (marveldo) [reply api/v1/models/testimonials.py:21]: sir please can you explain more on this

**Comment by marveldo** [comment]
Sir ive added necessary but couldnt change the jsonresponsedict and migrations due to me not understanding and some errors in jsondict

**Comment by marveldo** [comment]
@Dev-wonderful  ive effected the changes but icant use jsondict because im getting errors with it and cant change and is still in confusion about the migration

**Pull request closed without merging**